app/src/main/java/io/pslab/activity/OscilloscopeActivity.java

```diff
@@ -1092,69 +1090,34 @@ protected void onPostExecute(Void aVoid) {
     }
 
     public class XYPlotTask extends AsyncTask<String, Void, Void> {
-        String analogInput;
-        float[] xFloatData = new float[1000];
-        float[] yFloatData = new float[1000];
+        private String analogInput1;
+        private String analogInput2;
+        private float[] xFloatData;
+        private float[] yFloatData;
 
         @Override
         protected Void doInBackground(String... params) {
-            HashMap<String, double[]> data;
-            if ("CH2".equals(xyPlotXAxisChannel) || "CH2".equals(xyPlotYAxisChannel)) {
-                analogInput = params[0];
-                data = scienceLab.captureTwo(1000, 10, analogInput, false);
```

Comment:
Ideal method would be this. With current implementation it fetches one trace and then fetch the second trace after sometime; not simultaneous. CaptureTwo fetches them simultaneously.

Reply:
Okay. will make the change